aiohue/v2/__init__.py

```diff
@@ -246,3 +247,35 @@ async def fetch_full_state(self) -> None:
             self._sensors.initialize(full_state),
             self._groups.initialize(full_state),
         )
+
+    async def get_diagnostics(self) -> Dict[str, Any]:
+        """Return a dict with diagnostic information for debugging and support purposes."""
+        result = {}
+        subst_ids = {}
+
+        def _replace_id(item: dict):
+            """Replace ID with a random one for privacy reasons."""
+            id_key = "id" if "id" in item else "rid"
+            org_item_id = item[id_key]
+            if org_item_id not in subst_ids:
+                subst_ids[org_item_id] = f"redacted-{uuid4()}"
+            new_id = subst_ids[org_item_id]
+            item[id_key] = new_id
+
+        # add full state to result
+        full_state = await self.request("get", "clip/v2/resource")
+        for item in full_state:
+            _replace_id(item)
+        result["full_state"] = full_state
+
+        # add last event messages to result
+        last_events = []
+        for item in self._events.last_events:
+            # copy the object so we're not modifying the original
+            item = {**item}
+            if "id" in item or "rid" in item:
+                _replace_id(item)
```

Comment:
Only copy if we are mutating.
```suggestion
            if "id" in item or "rid" in item:
                item = {**item}
                _replace_id(item)
```